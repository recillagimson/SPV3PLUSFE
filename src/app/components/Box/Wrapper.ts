import styled from 'styled-components/macro';
import { media } from 'styles/media';
import { StyleConstants } from 'styles/StyleConstants';

/**
 * Main box wrapper
 * NOTE: this box will always take 100% width of the parent container
 *       if you want to resize this box, wrap it in a container with your specified width of choice
 *       ie: <div style={{width: 300px}}><Box>child</Box></div>
 */
const Wrapper = styled.div<{ children?: any }>`
  background-color: ${StyleConstants.WHITE};
  border-radius: ${StyleConstants.BORDER_RADIUS};
  margin: 15px auto;
  display: flex;
  align-items: flex-start;
  justify-content: stretch;
  flex-direction: row;
  flex-wrap: wrap;
  width: 100%;

  & > * {
    flex: 0 0 100%;
  }

  .bt-child {
    flex-grow: 1;
  }

  ${media.medium`
    max-width: 930px;
  `}

  ${media.large`
    max-width: 1000px;
  `}
`;

export default Wrapper;
