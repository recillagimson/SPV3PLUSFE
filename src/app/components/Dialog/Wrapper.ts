/**
 * Wrapper for the dialog component
 * this will have a fixed position on the page
 */
import styled from 'styled-components/macro';
import { media } from 'styles/media';
import { StyleConstants } from 'styles/StyleConstants';

const sizes = {
  small: '350px',
  medium: '640px',
  large: '960px',
  xlarge: '80%',
};

type DialogWrapperProps = {
  size?: 'small' | 'medium' | 'large' | 'xlarge' | undefined;
};

const Wrapper = styled.div<DialogWrapperProps>`
  position: fixed;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  width: 100%;
  height: 100%;
  background-color: ${StyleConstants.BLACK_TRANSPARENT_BG};
  z-index: 1000;
  display: flex;
  align-items: center;
  justify-content: center;

  .dialog-child {
    position: relative;
    z-index: 2;
    width: 100%;
    max-width: 95%;
    display: flex;
    flex-direction: column;
    background-color: #fff;
    border: 1px solid ${StyleConstants.BORDER_COLOR};
    border-radius: ${StyleConstants.BORDER_RADIUS};

    .dialog-content {
      flex: 1;
      padding: 20px;
    }
  }

  &.dialog-enter {
    opacity: 0;

    .dialog-child {
      opacity: 0;
      transform: scale(1.1);
    }
  }

  &.dialog-enter-active {
    opacity: 1;
    transition: opacity 200ms;

    .dialog-child {
      opacity: 1;
      transform: scale(1);
      transition: opacity 300ms, transform 300ms;
    }
  }

  &.dialog-exit {
    opacity: 1;

    .dialog-child {
      opacity: 1;
      transform: scale(1);
    }
  }

  &.dialog-exit-active {
    opacity: 0;
    transition: opacity 200ms;

    .dialog-child {
      opacity: 0;
      transform: scale(0.9);
      transition: opacity 300ms, transform 300ms;
    }
  }

  ${media.medium`
    .dialog-child {
      max-width: ${p => (p.size ? sizes[p.size] : '90%')};
    }
  `}
`;

export default Wrapper;
