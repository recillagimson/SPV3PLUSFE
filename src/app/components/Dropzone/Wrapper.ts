import styled from 'styled-components/macro';
import { StyleConstants } from 'styles/StyleConstants';

/**
 * Dropzone UI Wrapper
 */
const Wrapper = styled.div<{ bg?: boolean }>`
  background-color: ${p =>
    p.bg ? StyleConstants.LIGHT_GRAY_BG : 'transparent'};
  border-radius: ${StyleConstants.BORDER_RADIUS};
  border: 2px dashed ${StyleConstants.GOLD};
  width: 100%;
  margin: 0 0 20px;
  display: flex;
  align-items: center;
  justify-content: center;
  min-height: 250px;
  flex-direction: column;
  text-align: center;
  color: ${StyleConstants.LIGHT_GRAY_TEXT};
  position: relative;
  overflow: hidden;

  .svg-inline--fa {
    font-size: 2.5rem;
  }

  input {
    display: none;
    width: 0;
    height: 0;
    position: absolute;
    top: -999px;
    left: -999px;
  }

  .btn-upload {
    border: 0;
    background-color: transparent;
    padding: 0;
    color: ${StyleConstants.GOLD};
    font-size: inherit;
    cursor: pointer;
  }
`;

export default Wrapper;
