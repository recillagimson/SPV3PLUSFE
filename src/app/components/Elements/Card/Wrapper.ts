import styled from 'styled-components/macro';

type Props = {
  size?: 'small' | 'medium';
};

const sizes = {
  small: '50%',
  medium: '75%',
};

const Wrapper = styled.div<Props>`
  width: ${p => (p.size ? sizes[p.size] : '100%')};
  border: 1px solid #fff;
  border-radius: 3px;
  display: flex;
  flex-direction: column;
  background-color: white;
  .title {
    font-weight: bold;
    padding: 20px 20px;
    border-bottom: 1px solid #eee;
  }
  .body {
    padding: 20px 20px;
    flex: 1;
  }
  .footer {
    border-top: 1px solid #eee;
    padding: 15px 20px;
  }
`;

export default Wrapper;
