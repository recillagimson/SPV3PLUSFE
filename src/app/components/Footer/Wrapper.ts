import styled from 'styled-components/macro';
import { StyleConstants } from 'styles/StyleConstants';

const Wrapper = styled.footer`
  text-align: center;
  padding: 20px;
  font-size: 0.9em;

  p {
    margin: 0 0;
    color: #a9b1b8;
  }

  .authenticated & {
    margin-top: auto;
    border-top: 1px solid ${StyleConstants.BORDER_COLOR};
  }
`;

export default Wrapper;
