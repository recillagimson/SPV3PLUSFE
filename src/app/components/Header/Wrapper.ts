import styled from 'styled-components/macro';
import { media } from 'styles/media';
import { StyleConstants } from 'styles/StyleConstants';

import { Props } from './type';

const Wrapper = styled.header<Props>`
  position: fixed;
  top: 0;
  left: 0;
  z-index: 999;
  width: 100%;
  background-color: ${StyleConstants.WHITE};
  box-shadow: ${StyleConstants.BOX_SHADOW};
  padding: ${p => (p.isLoggedIn ? '15px 25px' : '20px')};

  .spdin & {
    position: sticky;
  }

  .wrapped {
    margin: 0 auto;
    display: flex;
    align-items: center;
  }

  .logo {
    width: 150px;
    align-self: flex-start;
  }

  ${media.medium`
    .wrapped {
      max-width: ${p => (!p.isLoggedIn ? StyleConstants.MAX_WIDTH : 'auto')};
    }
  `}
`;

export default Wrapper;
