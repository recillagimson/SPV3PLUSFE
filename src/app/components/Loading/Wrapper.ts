import styled from 'styled-components/macro';

interface Props {
  position?: string;
}

const Wrapper = styled.div<Props>`
  width: 100%;
  height: 100%;
  display: flex;
  align-items: center;
  justify-content: center;
  position: ${p => (p.position ? p.position : 'relative')};

  ${p =>
    p.position &&
    `
    top: 0;
    left: 0;
    right: 0;
    bottom: 0;
    background-color: rgba(255,255,255,.5);
  `}

  img {
    width: 300px;
  }
`;

export default Wrapper;
