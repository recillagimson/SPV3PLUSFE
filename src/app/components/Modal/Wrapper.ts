import styled from 'styled-components/macro';
import { StyleConstants } from 'styles/StyleConstants';

const sizes = {
  xsmall: '320px',
  small: '400px',
  medium: '640px',
  large: '960px',
  xlarge: '85%',
};

type ModalWrapperProps = {
  size?: 'xsmall' | 'small' | 'medium' | 'large' | 'xlarge';
  noPadding?: boolean;
};

const Wrapper = styled.div<ModalWrapperProps>`
  position: fixed;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  width: 100%;
  height: 100%;
  background-color: ${StyleConstants.transparent.black};
  z-index: 3000;
  display: flex;
  align-items: center;
  justify-content: center;

  .dialog-child {
    position: relative;
    z-index: 2;
    width: ${p => (p.size ? sizes[p.size] : '100%')};
    max-width: 95%;
    display: flex;
    flex-direction: column;
    background-color: #fff;
    /* border: 1px solid ${StyleConstants.divider}; */
    box-shadow: ${StyleConstants.shadow[2]};
    border-radius: ${StyleConstants.radius.large};

    .dialog-close {
      position: absolute;
      top: 16px;
      right: 10px;
    }

    .dialog-content {
      flex: 1;
      padding: ${p => (p.noPadding ? '0 0 0' : StyleConstants.spacing[24])};

      &.center {
        text-align: center;
      }
    }
  }

  &.dialog-enter {
    opacity: 0;

    .dialog-child {
      opacity: 0;
      transform: scale(1.1);
    }
  }

  &.dialog-enter-active {
    opacity: 1;
    transition: opacity 200ms;

    .dialog-child {
      opacity: 1;
      transform: scale(1);
      transition: opacity 300ms, transform 300ms;
    }
  }

  &.dialog-exit {
    opacity: 1;

    .dialog-child {
      opacity: 1;
      transform: scale(1);
    }
  }

  &.dialog-exit-active {
    opacity: 0;
    transition: opacity 200ms;

    .dialog-child {
      opacity: 0;
      transform: scale(0.9);
      transition: opacity 300ms, transform 300ms;
    }
  }
`;

export default Wrapper;
