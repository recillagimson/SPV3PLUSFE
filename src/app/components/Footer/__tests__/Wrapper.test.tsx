import React from 'react';
import Wrapper from '../Wrapper';
import { render } from '@testing-library/react';

describe('<Wrapper />', () => {
  it('renders correctly', () => {
    const tree = render(<Wrapper />).baseElement;
    expect(tree).toMatchSnapshot();
  });
});
