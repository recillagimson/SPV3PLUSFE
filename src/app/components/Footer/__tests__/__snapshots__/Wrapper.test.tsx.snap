// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Wrapper /> renders correctly 1`] = `
.c1 {
  text-align: center;
  padding: 20px;
  font-size: 0.9em;
}

.c1 p {
  margin: 0 0;
  color: #a9b1b8;
}

.c1 p .small {
  display: inline-block;
  font-size: 0.8em;
  opacity: 0.75;
  padding-left: 5px;
}

.authenticated .c0 {
  margin-top: auto;
  border-top: 1px solid #CED4DA;
}

<body>
  <div>
    <footer
      class="c0 c1"
    />
  </div>
</body>
`;
